### DISCUSSION TITLE
Use cowforest as the default backend for bridgenodes

### DISCUSSION THREAD

**Discussion by kcalvinalvin** [mit-dci/utreexo#238] [Ideas]
Cowforest still has some things to look at (caching the treetables, what height to set the treeblock to, etc) but I think it's better than all other disk-based options and ramForest is really not doable unless you have 20GB+ of ram.

In the coming weeks, we should make improvements to cowforest and make it default. What does everyone else think?

**Comment by adiabat** [comment]
Yeah if it isn't buggy and the performance is good we can default to cow.  

I also wonder why in ram takes so much ram.  It shouldn't need more than 8GB, and it's straightforward enough that I'm not sure where the extra ram is going.

2^27 = 134M (there are just over 70M utxos) and 32 bytes * 2^27 * 2 = 2^33, or 8GB.

I guess you need a bit more ram for other stuff, but 16GB should be fine, which is a normal-ish amount for good computers these days.

**Reply by dergoegge** [reply]
The position map is also quite big. (12+8)*2^27 = ~2.6GB

But 16GB should still be enough...

**Comment by dergoegge** [comment]
I think we can make it the default if we add some more tests and maybe figure out the optimal value for the treeblock size.

(I still like the ram forest the most since its simple and fast if your machine can handle it but the cow forest was the only forest type that worked on the test server.)

**Reply by kcalvinalvin** [reply]
> I think we can make it the default if we add some more tests and maybe figure out the optimal value for the treeblock size.

Ok yeah, I'll put that on the TODO list for myself and get to it after the btcd implementation is somewhat complete.

> (I still like the ram forest the most since its simple and fast if your machine can handle it but the cow forest was the only forest type that worked on the test server.)

Heh yeah I also don't use cow for when I'm testing stuff since I want the results as quick as possible.
